tests/scene/test_visual_shader.h

```diff
@@ -0,0 +1,151 @@
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2022 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2022 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef TEST_VISUAL_SHADER_H
+#define TEST_VISUAL_SHADER_H
+
+#include "scene/resources/visual_shader.h"
+
+#include "tests/test_macros.h"
+
+namespace TestVisualArray {
+
+TEST_CASE("[SceneTree][VisualShader] Object creation and parameter") {
+	Ref<VisualShader> vs = memnew(VisualShader);
+	CHECK(vs.is_valid());
+
+	CHECK(vs->get_mode() == Shader::MODE_SPATIAL);
+
+	for (int i = 1; i < Shader::MODE_MAX; i++) {
+		vs->set_mode((Shader::Mode)i);
+		CHECK(vs->get_mode() == i);
+	}
+}
+
+TEST_CASE("[SceneTree][VisualShader] Testing VisualShaderNodes") {
+	SUBCASE("Testing Node Creation") {
+		Ref<VisualShader> vs = memnew(VisualShader);
+		CHECK(vs.is_valid());
+
+		for (int i = 0; i < VisualShader::TYPE_MAX; i++) {
+			Ref<VisualShaderNode> vsn = memnew(VisualShaderNodeInput);
+			CHECK(vsn.is_valid());
+			vs->add_node(VisualShader::Type(i), vsn, Vector2(1, 10), i + 2);
+			CHECK(vs->get_node(VisualShader::Type(i), i + 2) == vsn);
+		}
+
+		ERR_PRINT_OFF;
+
+		// Testing for Invalid entries
```

Comment:
```suggestion
		// Testing for Invalid entries.
```

The comments should be ended with `.`.